### PR TITLE
Move framework testing libraries to the right place

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind cleanup
/sig scheduling

#### What this PR does / why we need it:

The motivation for this pr is because https://github.com/kubernetes/kubernetes/issues/120002, where we want to replace the `makeBasePod` with `podWrapper` to reduce the maintenance cost. However, we have the import cycle as

`pkg/scheduler/testing` -> `pkg/scheduler/framework` -> `pkg/scheduler/testing` if we want to use the `podWrapper`.

Considering part of the files under `pkg/scheduler/testing` are all related to the framework, we can move them under the `pkg/scheduler/framework/testing`. 

What we do here is:
1. Rename `pkg/scheduler/framework/fake` -> `pkg/scheduler/framework/testing`
2. Rename `listers.go` -> `fake_listers.go` as the folder name changed from `fake` -> `testing`
3. Move `fake_extender.go`, `fake_plugins.go`, `framework_helpers.go` from `pkg/scheduler/testing` -> `pkg/scheduler/framework/testing`

No logic changes at all.
cc @Huang-Wei who introduced the change 3 years ago for advices. https://github.com/kubernetes/kubernetes/pull/92107

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Jerry-Kon** [comment]
Preparing pr description...

**Comment by kerthcet** [comment]
@alculquicondor can you take a look of this pr?

**Comment by Huang-Wei** [comment]
I can take a look.

/assign

**Review by Huang-Wei — COMMENTED** [review]
Can we hold all testing bits under `pkg/scheduler/testing`? say:

- pkg/scheduler/testing
- pkg/scheduler/testing/framework
- pkg/scheduler/testing/fake (if needed)

So developers just need to keep this single testing pkg in mind to search existing utilities.

**Comment by kerthcet** [comment]
> Can we hold all testing bits under `pkg/scheduler/testing`? say:
> 
> * pkg/scheduler/testing
> * pkg/scheduler/testing/framework
> * pkg/scheduler/testing/fake (if needed)
> 
> So developers just need to keep this single testing pkg in mind to search existing utilities.

Make sense.

File: pkg/scheduler/testing/framework/fake_extender.go
Comment (Huang-Wei) [inline pkg/scheduler/testing/framework/fake_extender.go:1]: need to update the `package [xyz]` literals (applies elsewhere)

File: pkg/scheduler/testing/framework/fake_listers.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package fake
+package testing
```
Comment (Huang-Wei) [inline pkg/scheduler/testing/framework/fake_listers.go:17]:
```suggestion
package framework
```

Comment (kerthcet) [reply pkg/scheduler/testing/framework/fake_listers.go:17]: Done.

**Comment by alculquicondor** [comment]
Don't wait for my review.

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks!

**Comment by Huang-Wei** [comment]
cc @msau42 for approving changes in `pkg/controller/volume/`.

**Pull request merged**